### ISSUE TITLE
Deprecate the redundant assertion method aliases on `twisted.trial.unittest.TestCase`

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4991]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4991|
|Type|enhancement|
|Created|2011-03-26 19:38:08Z|

Per the mailing list discussion, [TestCase methods names](http://twistedmatrix.com/pipermail/twisted-python/2011-March/023681.html), after [#4989](https://github.com/twisted/twisted/issues/4989) and [#4990](https://github.com/twisted/twisted/issues/4990) are resolved, we should deprecate `assertEquals`, `failUnlessEquals`, and so on, leaving just the `assertEqual` style of methods.

Attachments:

* [deprecate-fail-methods.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/5ea/5ea4cdaef6d8edfe44b7f29c1db824ab4bc3cdd5/714020c209736caa7bcebb048f068cb8a90e8ee1.patch) (20216 bytes) - added by Julian on 2012-07-09 01:02:27Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4991 4991
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__new new
resolution__None None
component__trial trial
keywords__ 
time__1301168288000000 1301168288000000
changetime__1363665825000000 1363665825000000
version__None None
owner__ 
cc__jml cc__Julian
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_2"></a>|@jyknight commented|
|-|-|

Also remove failIf*, failUnless*, and assert_, right?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Deprecate for now, yes.  We should end up with a single preferred spelling for each kind of assertion.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian commented|
|-|-|

Attached a patch that does the deprecations. I had a bit of trouble with `twisted.python.deprecate` -- the tests I wrote fail because deprecate produces for my modifications things like "twisted.trial.unittest.failIf was deprecated in Twisted 13.0.0" whereas callDeprecated is expecting a fully qualified path to the TestCase being run. Hoping someone can give me a pointer there on what needs modifying.

I'll open another ticket for fixing the test suite to not use the deprecated aliases, unless someone speaks up and prefers it be done here.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

> I'll open another ticket for fixing the test suite to not use the deprecated aliases, unless someone speaks up and prefers it be done here.

That ticket already exists, [#4990](https://github.com/twisted/twisted/issues/4990), and needs to be resolved before this one.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Uhh.  It seems it is "resolved" already, even though the issue it describes is not actually fixed.  I guess that means we do need another ticket.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince commented|
|-|-|

A couple of review points, for whoever rebases this patch.

1. It would be better to keep the alternate names beside the canonical name.

1. The tests that explicitly call deprecate methods should explicitly handle the warning, so that running with warnings enabled doesn't spuriously report them.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_9"></a>|@Julian commented|
|-|-|

For reference, the list of methods that this ticket will aim to deprecate attempt to leave one preferred spelling for each assertion. The preferred spelling is the one preferred by pyunit where possible. Therefore, the deprecations are as follows:

fail*(, which are all provided as assert*( methods
*Equals(, which are provided by *Equal(
assert_ -> assertTrue
assertNot -> assertFalse
assertIdentical -> assertIs
assertNotIdentical -> assertIsNot
assertSubstring -> assertIn
assertNotSubstring -> assertNotIn

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_10"></a>|@Julian commented|
|-|-|

Argh, sorry, maybe in a few years I will learn how to WikiFormat. Dup-posting for readability.

For reference, the list of methods that this ticket will aim to deprecate attempt to leave one preferred spelling for each assertion. The preferred spelling is the one preferred by pyunit where possible. Therefore, the deprecations are as follows:

* fail*(, which are all provided as assert*( methods
* *Equals(, which are provided by *Equal(
* assert_ -> assertTrue
* assertNot -> assertFalse
* assertIdentical -> assertIs
* assertNotIdentical -> assertIsNot
* assertSubstring -> assertIn
* assertNotSubstring -> assertNotIn

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince commented|
|-|-|

:)
 * fail*(, which are all provided as assert*( methods
 * *Equals(, which are provided by *Equal(
 * assert_ -> assertTrue
 * assertNot -> assertFalse
 * assertIdentical -> assertIs
 * assertNotIdentical -> assertIsNot
 * assertSubstring -> assertIn
 * assertNotSubstring -> assertNotIn 

(bullets and numbered list need a leading space)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_12"></a>|@tomprince commented|
|-|-|

Note that twisted doesn't actually have `assertIs`.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_13"></a>|@Julian commented|
|-|-|

Tickets [#6220](https://github.com/twisted/twisted/issues/6220) - [#6233](https://github.com/twisted/twisted/issues/6233) need to be done first before this can happen.
